java/semantickernel-core/src/main/java/com/microsoft/semantickernel/templateengine/DefaultPromptTemplateEngine.java

```diff
@@ -25,7 +28,7 @@
 // immutable when invoked within the template
 /// </summary>
 public class DefaultPromptTemplateEngine implements PromptTemplateEngine {
-    // private readonly ILogger _log;
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultPromptTemplateEngine.class);
```

Comment:
Its an unusual pattern for java to inject loggers, general standard practice is to have loggers as statics, is there a usecase that makes injecting loggers important? Due to the lack of default arguments, this will mean you will end up either passing null around a lot when constructing objects, or increasing the number of constructors for all of our objects to add an optional logger.

Reply:
I chatted with Shawn about this and no change required at this time. I'm keeping track of where the implementation diverges from .Net and we may discuss this again when we're ready to merge to main.